src/sasl_plain.c

```diff
@@ -66,8 +67,17 @@ CONCRETE_SASL_MECHANISM_HANDLE saslplain_create(void* config)
                 else
                 {
                     /* Ignore UTF8 for now */
-                    result->init_bytes = (unsigned char*)malloc(authzid_length + authcid_length + passwd_length + 2);
-                    if (result->init_bytes == NULL)
+                    //size_t malloc_size = authzid_length + authcid_length + passwd_length + 2);  // use safe add
+                    size_t malloc_size = safe_add_size_t(authzid_length, authcid_length);
+                    malloc_size = safe_add_size_t(malloc_size, passwd_length);
+                    malloc_size = safe_add_size_t(malloc_size, 2);
+
+                    if (malloc_size == SIZE_MAX)
```

Comment:
I have similar comments as in c-util about disregarding SIZE_MAX as a valid size, but will refrain from posting all over because we are intentionally choosing to do so.

Reply:
As discussed yesterday. safe_add is a clamping function. mallocing all memory is an error condition.